File: 08/spring/bullet_hole.py
def bullet_hole(a, b, c, vx, vy, vz):
    try:
        t = 1.0*c / (a*vx + b*vy)
        if t < 0: return "Travels away from plane"
        return "Impacts after %.2f seconds at position (%.2f, %.2f, %.2f)" % \
                (t, vx*t, vy*t, vz*t - 9.8*t**2 / 2.0)
    except ZeroDivisionError:
        return "Travels parallel to plane"

if __name__ == '__main__':
    while 1:
        try:
            line1 = raw_input()
            line2 = raw_input()
            raw_input()
        except: break
        vx, vy, vz = map(int, line1.split())
        a, b, c = map(int, line2.split())
        print bullet_hole(a, b, c, vx, vy, vz)

File: 08/spring/solutions/bullet_hole.java
import java.util.Scanner;

class bullet_hole {

	private static int vx, vy, vz;

	private static double x(double t) {
		return vx*t;
	}

	private static double y(double t) {
		return vy*t;
	}

	private static double z(double t) {
		return (vz - (4.9*t))*t;
	}

	public static void main(String[] arguments) {

		Scanner in = new Scanner(System.in);

		while (in.hasNextInt()) {

			vx = in.nextInt();
			vy = in.nextInt();
			vz = in.nextInt();

			int a = in.nextInt();
			int b = in.nextInt();
			int c = in.nextInt();

			int denom = a*vx + b*vy;

			if (denom == 0)
				System.out.printf("Travels parallel to plane%n");
			else {
				double t = (double)c / denom;

				if (t < 0) {
					System.out.printf("Travels away from plane%n");
				}
				else {
					System.out.printf("Impacts after %.2f seconds at position (%.2f, %.2f, %.2f)%n", t, x(t), y(t), z(t));
				}
			}
		}
	}
}
